### ISSUE TITLE
Concurrency exception in DbOffsetStore

### ISSUE THREAD

**Issue by BEagle1984** [BEagle1984/silverback#96]
Sometimes the `DbOffsetStore` throws a `DbUpdateConcurrencyException` when committing.
(The error isn't even handled by the policies and the consumer stops.)
This happens after having upgraded to the version 2.1.1.

```
Fatal error occurred consuming the message 286783 from endpoint *******. The consumer will be stopped.
Microsoft.EntityFrameworkCore.DbUpdateConcurrencyException: Database operation expected to affect 1 row(s) but actually affected 0 row(s). Data may have been modified or deleted since entities were loaded. See http: //go.microsoft.com/fwlink/?LinkId=527962 for information on understanding and handling optimistic concurrency exceptions.
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ThrowAggregateUpdateConcurrencyException(Int32 commandIndex, Int32 expectedRowsAffected, Int32 rowsAffected)
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ConsumeResultSetWithPropagationAsync(Int32 commandIndex, RelationalDataReader reader, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ConsumeAsync(RelationalDataReader reader, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Silverback.EntityFrameworkCore.DbContextEventsPublisher.ExecuteSaveTransaction(Func`1 saveChanges, Boolean async)
   at Silverback.EntityFrameworkCore.DbContextEventsPublisher.ExecuteSaveTransaction(Func`1 saveChanges, Boolean async)
   at Silverback.Messaging.Connectors.Repositories.DbOffsetStore.Commit()
   at Silverback.Util.EnumerableForEachExtensions.ForEachAsync[T](IEnumerable`1 source, Func`2 action)
   at Silverback.Util.TaskExtensions.GetReturnValue(Task task)   at Silverback.Util.EnumerableSelectExtensions.<>c__DisplayClass2_0`2.<<SelectAsync>b__0>d.MoveNext()
```

**Comment by BEagle1984** [comment]
Tried many combinations but still unable to reproduce.
No concurrency check is enabled on the stored offsets `DbSet`, therefore such error doesn't popup even manually tampering with the data.

I guess that the error is caused by something wrong in a subscriber that causes the `SaveChanges` in the `DbOffsetStore` to throw the exception. (It is maybe related to a missing or not awaited `SaveChanges` or a swallowed exception?)

Some further investigation is necessary. I'll check if it's possible to get to debug the failing code.

**Comment by BEagle1984** [comment]
Fixed in v3.0.0 (unable to reproduce)
